cloud/wfm/src/task/BaseTask.ts

```diff
@@ -0,0 +1,64 @@
+import { EventEmitter } from 'eventemitter3';
+import {Result} from '../result';
+import { Task, TaskEventData, TaskStatus } from './index';
+
+/**
+ * Base implementation for {@link Task}s
+ * Derived classes are expected to implement their run {@link #run} methods
+ */
+class BaseTask extends EventEmitter implements Task {
+  public result: Result;
+  /**
+   * Storage field for the `status` property
+   */
+  protected _status: TaskStatus | number = TaskStatus.PENDING;
+  protected options: object;
+
+  /**
+   * Setter for the status property
+   * Emits a statusChange event when set
+   */
+  public set status(to: TaskStatus | number) {
+    if (to === this._status) {
+      return;
```

Comment:
This type of code can mask possible problems in developer code.
Worth to use logger here once it will be available ;)